t/joi.t

```diff
@@ -118,6 +118,36 @@ is_deeply(
   'extended object',
 );
 
+is_deeply(
+  edj(
+    joi->object->props(x => joi->integer, y => joi->integer->required)
+      ->extend(
+        joi->object->props(
+            k => joi->object->props(
+                x => joi->string->required,
+                y => joi->string->required
+            )
+        )
+     )
+  ),
+  {
+    type       => 'object',
+    properties => {
+      x => {type => 'integer'},
+      y => {type => 'integer'},
+      k => {
+        type => 'object',
+        properties => {
+          x => {type => 'string'}, y => {type => 'string'},
+        },
+        required => ['x', 'y']
+      }
+    },
+    required   => ['k','y'],
+  },
+  'extended object',
+);
```

Comment:
This test is not testing what I think you intend it to test.

Reply:
hmm 
if comment ```
push @{ $clone->{required} }, @{ $self->{required} }
    if $self->{required} && @{$self->{required}};
```
test fail

Reply:
bag... got it )) thanks

Reply:
Yes, but your structure has a nested object, which is not touched by the merging. extend() only works on the first level.